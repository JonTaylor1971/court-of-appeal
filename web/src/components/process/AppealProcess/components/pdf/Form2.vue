<template>
<div v-if="dataReady">  

    <b-row >
        <b-col cols="10">
            <b-button 
                style="float: right;" 
                variant="primary"
                @click="navigateToEditPage()"
                ><b-icon-pencil-square class="mx-0" variant="white" scale="1" ></b-icon-pencil-square>
                Go Back and Edit
            </b-button>
        </b-col>
        <b-col cols="2">
            <b-button
                style="float: right;" 
                variant="success"
                @click="onPrint()"
                >Download PDF
                <b-icon-printer-fill class="mx-0" variant="white" scale="1" ></b-icon-printer-fill>
            </b-button>
        </b-col>
    </b-row>  
  
    <b-card id="print" style="border:1px solid; border-radius:5px;" bg-variant="white" class="mt-4 mb-4 container" no-body>
        <form-2-layout v-bind:result="result"/>
    </b-card>
</div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import { namespace } from "vuex-class";
import "@/store/modules/information";
import { form2DataInfoType } from '@/types/Information';
const informationState = namespace("Information");
import Form2Layout from "./Form2Layout.vue";
import moment from 'moment';

@Component({
    components:{        
        Form2Layout
    }
})
export default class Form2 extends Vue {

    @Prop({required: true})
    applicationId!: string;
    
    @informationState.State
    public form2Info: form2DataInfoType;

    result = {} as form2DataInfoType;
    dataReady = false;
   
    mounted(){
        this.dataReady = false;
        this.result = this.getForm2Data();       
        this.dataReady = true;
        Vue.nextTick(()=> this.onPrint())
    }   
           
    public onPrint() { 
        const pdf_type = Vue.filter('getPathwayPdfType')("priorityParenting")
        const pdf_name = "form2-" + this.applicationId;
        const el= document.getElementById("print");

      
        const bottomLeftText = ``;
        const bottomRightText = `" "`
        const url = '/survey-print/'+this.applicationId+'/?name=' + pdf_name + '&pdf_type='+pdf_type+'&version=1.0&noDownload=true'
        const pdfhtml = Vue.filter('printPdf')(el.innerHTML, bottomLeftText, bottomRightText );

        const body = {
            'html':pdfhtml,
            'json_data':this.result
        }       
        
        const options = {
            responseType: "blob",
            headers: {
            "Content-Type": "application/json",
            }
        }  

        this.$http.post(url,body, options)
        .then(res => {
            const currentDate = moment().format();
            this.$store.commit("Application/setLastPrinted", currentDate); 
            
                
        },err => {
            console.error(err);        
        });
    }

    public navigateToEditPage() {
        this.$router.push({name: "fill" })
    }
 
    public getForm2Data() {   
        
        //TODO: use api GET call to gather data
        return this.form2Info;
    }
}
</script>
<style scoped lang="scss" src="@/styles/_pdf.scss">

</style>