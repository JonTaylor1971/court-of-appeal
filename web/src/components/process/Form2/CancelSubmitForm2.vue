<template>
    <b-card v-if="mountedData" header-tag="header" bg-variant="light" border-variant="white" style="width: 80rem;" class="mx-auto">

        <b-card-header header-bg-variant="light">            
            <form-2-process-header v-bind:stepsCompleted="stepsCompleted"/>
        </b-card-header>

        <b-card no-body bg-variant="light" border-variant="light" class="my-2 text-dark">

            <b-row class="ml-5 mt-2 mb-4" style="font-size: 2rem;">
                E-Filing Cancelled
            </b-row>

            <b-card no-body class="border-white bg-white mx-4"> 
                <b-row class="ml-5 my-3" style="font-size: 14px;">
                    E-Filing has been cancelled.
                </b-row>

                <b-row class="mt-1 mb-3 mr-5">
                    <b-col cols="12">
                        <b-button 
                            style="float: right;" 
                            variant="success"
                            @click="done()"
                            >
                            Done
                        </b-button>
                    </b-col>                
                </b-row>
            </b-card>

        </b-card>

        
    </b-card>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import Form2ProcessHeader from "@/components/process/Form2/components/Form2ProcessHeader.vue";
import { form2StatusInfoType } from '@/types/Information';

@Component({
    components:{
        Form2ProcessHeader
    }
})
export default class CancelSubmitForm2 extends Vue {

    stepsCompleted = {} as form2StatusInfoType;  
    mountedData = false;     

    mounted() {
        console.log('here')
        this.mountedData = false;        

        this.stepsCompleted = {
            first: true,
            second: true,
            third: false
        }
        this.mountedData = true;                   
    }    

    public done() {
        this.$router.push({name: "dashboard" }) 
    }
}
</script>

<style scoped lang="scss">



</style>