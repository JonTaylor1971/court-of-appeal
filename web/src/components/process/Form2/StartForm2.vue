<template>
    <b-card header-tag="header" bg-variant="light" border-variant="white" style="width: 80rem;" class="mx-auto">

        <!-- <b-card bg-variant="white" border-variant="white">            
            <form-2-process-header v-bind:stepsCompleted="stepsCompleted"/>
        </b-card> -->

        <template #header bg-variant="white" border-variant="white">            
            <form-2-process-header v-bind:stepsCompleted="stepsCompleted"/>
        </template>

        <b-card border-variant="light" bg-variant="light" class="my-2">

            <div class="ml-5" style="font-size: 2rem;">
                Notice of Appearance
            </div> 
            <div class="ml-5" style="font-size: 14px;">
                Form 2
            </div>           
           
        </b-card>

        <b-card border-variant="white" bg-variant="white" class="mt-3 bg-white">

            <form-2-case-information/>
            
            
        </b-card>       
        
    </b-card>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import { namespace } from "vuex-class";
import "@/store/modules/information";
const informationState = namespace("Information");

import Form2ProcessHeader from "@/components/process/Form2/components/Form2ProcessHeader.vue";
import Form2CaseInformation from "@/components/process/Form2/components/Form2CaseInformation.vue";
import { form2StatusInfoType } from '@/types/Information';

@Component({
    components:{
        Form2ProcessHeader,
        Form2CaseInformation
    }
})
export default class StartForm2 extends Vue {

    stepsCompleted = {} as form2StatusInfoType;   

    mounted() {  
        this.stepsCompleted = {
            first: false,
            second: false,
            third: false
        }       
    }  


}
</script>

<style scoped lang="scss">



</style>