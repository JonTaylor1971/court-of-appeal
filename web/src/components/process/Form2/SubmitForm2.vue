<template>
    <b-card v-if="mountedData" header-tag="header" bg-variant="light" border-variant="white" style="width: 80rem;" class="mx-auto">

        <b-card-header header-bg-variant="light">            
            <form-2-process-header v-bind:stepsCompleted="stepsCompleted"/>
        </b-card-header>        

        <b-card no-body v-if="submitted" bg-variant="light" border-variant="light" class="my-2 text-dark">

            <b-row class="ml-5 mb-4 mt-2" style="font-size: 2rem;">
                Form Submitted
            </b-row> 
            
            <b-card no-body class="border-white bg-white mx-4">
                <b-row class="ml-5 my-4" style="font-size: 14px;">
                    Your Notice of Appearance has been submitted and accepted.
                </b-row>
                <b-row class="ml-5">
                    <b-col cols="2">
                        <span style="float: left; font-weight: 700">
                            Package Number:                        
                        </span>
                    </b-col>
                    <b-col cols="2">
                        <span style="float: left;">
                            {{packageInfo.packageNumber}}                            
                        </span>
                    </b-col>
                </b-row>

                <b-row class="ml-5">
                    <b-col cols="2">
                        <span style="float: left; font-weight: 700">
                            Your File #:                        
                        </span>
                    </b-col>
                    <b-col cols="2">
                        <span style="float: left;">
                            {{packageInfo.fileNumber}}                            
                        </span>
                    </b-col>
                </b-row>

                <b-row class="ml-5">
                    <b-col cols="2">
                        <span style="float: left; font-weight: 700">
                            Submitted To:                        
                        </span>
                    </b-col>
                    <b-col cols="2">
                        <span style="float: left;">
                            BC Court of Appeal                            
                        </span>
                    </b-col>
                </b-row>

                <b-row class="ml-5">
                    <b-col cols="2">
                        <span style="float: left; font-weight: 700">
                            E-Filing Url:                        
                        </span>
                    </b-col>
                    <b-col cols="2">
                        <span style="float: left;">
                            <a :href="packageInfo.eFilingUrl"
                                target="_blank">{{packageInfo.packageNumber}}</a>                                                        
                        </span>
                    </b-col>
                </b-row>

                <b-row class="my-5">
                    <b-col cols="10">
                        <b-button 
                            style="float: right;" 
                            variant="success"
                            @click="cancel()"
                            >
                            Done
                        </b-button>
                    </b-col>                
                </b-row>
            </b-card>           

        </b-card>

        <b-card v-else text-variant="dark" class="my-2 mx-5 bg-light border-light">

            <b-row class="ml-2" style="font-size: 2rem;">
                Submit through E-Filing
            </b-row> 
            <b-row class="ml-2 mb-5" style="font-size: 14px;">
                By submitting, you will be redirected to the E-Filing Hub.
            </b-row>

            <b-row class="ml-5">
                <b-col cols="10">
                    <b-button 
                        style="float: right;" 
                        variant="danger"
                        @click="cancel()"
                        >
                        Cancel
                    </b-button>
                </b-col>
                <b-col cols="2">
                    <b-button
                        style="float: left;" 
                        variant="success"
                        @click="submit()"
                        >Submit
                        <span class="fa fa-paper-plane btn-icon-left"/>
                    </b-button>
                </b-col>
            </b-row>

        </b-card>

        
    </b-card>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import Form2ProcessHeader from "@/components/process/Form2/components/Form2ProcessHeader.vue";
import { form2StatusInfoType, packageInfoType } from '@/types/Information';

@Component({
    components:{
        Form2ProcessHeader
    }
})
export default class SubmitForm2 extends Vue {

    stepsCompleted = {} as form2StatusInfoType;  
    mountedData = false; 
    submitted = false;
    packageInfo = {} as packageInfoType;

    mounted() {
        this.mountedData = false;
        // TODO: figure out if submitting or submitted
        if (this.submitted){
            this.packageInfo.packageNumber = '15140';
            this.packageInfo.fileNumber = '66';
            this.packageInfo.eFilingUrl = 'https://dev.justice.gov.bc.ca/efilinghub/packagereview/15140?packageNo=15140';

            this.stepsCompleted = {
                first: true,
                second: true,
                third: true
            } 
            this.saveApplication(this.packageInfo);


        } else {

            this.stepsCompleted = {
                first: true,
                second: true,
                third: false
            }
            this.mountedData = true;  

        }             
    }

    public saveApplication(newPackageInfo: packageInfoType) {
        //TODO: add this endpoint
        const data = {
            packageNumber: newPackageInfo.packageNumber,
            packageUrl: newPackageInfo.eFilingUrl 
        } 

        const url = "/efiling/"+newPackageInfo.fileNumber+"/submit/";

        const header = {
            responseType: "json",
            headers: {
                "Content-Type": "application/json",
            }
        }

        this.$http.put(url, data, header)
        .then(res => {                            
            this.mountedData = true;
        }, err => {
            console.error(err);
            this.mountedData = true;
        });    
    }    

    public submit() {
        console.log('submit-efiling');
    }

    public cancel() {
        this.$router.push({name: "dashboard" }) 
    }
}
</script>

<style scoped lang="scss">



</style>