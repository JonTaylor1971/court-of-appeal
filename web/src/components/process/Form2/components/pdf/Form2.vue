<template>
<div v-if="dataReady">  

    <b-row >
        <b-col cols="10">
            <b-button 
                style="float: right;" 
                variant="primary"
                @click="navigateToEditPage()"
                ><b-icon-pencil-square class="mx-0" variant="white" scale="1" ></b-icon-pencil-square>
                Go Back and Edit
            </b-button>
        </b-col>
        <b-col cols="2">
            <b-button
                style="float: right;" 
                variant="success"
                @click="savePdf()"
                >Download PDF
                <b-icon-printer-fill class="mx-0" variant="white" scale="1" ></b-icon-printer-fill>
            </b-button>
        </b-col>
    </b-row>  
  
    <b-card id="print" style="border:1px solid; border-radius:5px;" bg-variant="white" class="mt-4 mb-4 container" no-body>
        <form-2-layout v-bind:result="result"/>
    </b-card>
</div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import { namespace } from "vuex-class";
import "@/store/modules/information";
import { form2DataInfoType } from '@/types/Information';
const informationState = namespace("Information");
import Form2Layout from "./Form2Layout.vue";
import moment from 'moment';

@Component({
    components:{        
        Form2Layout
    }
})
export default class Form2 extends Vue {

    @Prop({required: true})
    caseId!: string;
    
    @informationState.State
    public currentCaseId: string;

    @informationState.Action
    public UpdateForm2Info!: (newForm2Info: form2DataInfoType) => void

    result = {} as form2DataInfoType;
    dataReady = false;
   
    mounted(){
        this.dataReady = false;
        this.getForm2Data(); 
    }   
           
    public onPrint() { 
        const pdf_type = "FORM2"
        const pdf_name = "form2-" + this.caseId;
        const el= document.getElementById("print");

      
        const bottomLeftText = ``;
        const bottomRightText = `" "`;        
        const url = '/form-print/'+this.caseId+'/?name=' + pdf_name + '&pdf_type='+pdf_type+'&version=1.0&noDownload=true'
        const pdfhtml = Vue.filter('printPdf')(el.innerHTML, bottomLeftText, bottomRightText );

        const body = {
            'html':pdfhtml,
            'json_data':this.result
        }       
        
        const options = {
            responseType: "blob",
            headers: {
            "Content-Type": "application/json",
            }
        }  

        this.$http.post(url,body, options)
        .then(res => {                       
                
        },err => {
            console.error(err);        
        });
    }

    public savePdf(){        
        const pdfType = "FORM2"
        const url = '/form-print/'+this.caseId+'/?pdf_type='+pdfType
        const options = {
            responseType: "blob",
            headers: {
            "Content-Type": "application/json",
            }
        }
        this.$http.get(url, options)
        .then(res => {
            const blob = res.data;
            const link = document.createElement("a");
            link.href = URL.createObjectURL(blob);
            document.body.appendChild(link);
            link.download = pdfType+".pdf";
            link.click();
            setTimeout(() => URL.revokeObjectURL(link.href), 1000);          

        },err => {
            console.error(err);
        });
    }

    public navigateToEditPage() {
        this.$router.push({name: "fill" })
    }
 
    public getForm2Data() {        
       
        this.$http.get('/case/'+this.currentCaseId+'/')
        .then((response) => {
            if(response?.data?.data){            
                            
                this.result = response.data.data
                this.UpdateForm2Info(this.result)                         
                this.dataReady = true;
                Vue.nextTick(()=> this.onPrint())
            }
                
        },(err) => {
        console.log(err)        
        });      
    }
}
</script>
<style scoped lang="scss" src="@/styles/_pdf.scss">

</style>