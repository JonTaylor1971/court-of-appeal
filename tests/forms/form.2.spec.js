require('../support/fake.dom');
import React from 'react';
import Form2 from '../../src/forms/Form.2';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });import renderer from "react-test-renderer";

test('Snapshots of form2', ()=> {

    const form2 = renderer.create(
        <Form2 id="form-2"
              service={{ createForm2: ()=> {} }}
        />,
    );
    let tree = form2.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = form2.getInstance();
    expect(instance.state.showForm2).toBe(false);

    instance.state.showForm2 = true;

    let e = { target: { name: "number.form-seven", value: "CA12345" }};
    instance.handleSearchFieldChange(e);
    expect(instance.state.formSevenNumber).toMatch('CA12345');

});
