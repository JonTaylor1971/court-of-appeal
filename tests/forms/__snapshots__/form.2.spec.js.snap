// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots of form2 1`] = `
<div
  className="template container gov-container form"
  id="topicTemplate"
>
  <div
    id="breadcrumbContainer"
  >
    <ol
      className="breadcrumb"
    >
      <li>
        <a
          href="/"
          id="home"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href=""
        >
          Notice of Appearance (Form 2)
        </a>
      </li>
    </ol>
  </div>
  <div
    className="row"
  >
    <div
      className="contentPageMainColumn col-sm-12"
      id="main-content"
      role="main"
    >
      <div
        className="form-title not-printable"
      >
        <h1>
          NOTICE OF APPEARANCE
        </h1>
        Form 2 (RULES 5 (A), 13 (A) AND 17 (A))
      </div>
      <div
        className="form-section not-printable"
      >
        <h2
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          When Do You Use the Notice of Appearance (Form 2)?
        </h2>
        <p>
          If you have been served a Notice of Appeal (Form 7), and you want to respond, fill out the Notice of Appearance (Form 2). If you do not respond, you will not be included in future court proceedings related to this case, and will not have access to the case documents filed.
        </p>
        <p>
          You have 
          <span
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            10 days
          </span>
           to serve a filed copy of the Notice of Appearance to the appellant, after receiving a Notice of Appeal.
        </p>
      </div>
      <div
        className="form-section not-printable"
      >
        <h2
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Appeal Case Information
        </h2>
        Enter the Court of Appeal number you want to respond to:
        <table>
          <tbody>
            <tr>
              <td>
                LEVEL OF COURT
              </td>
              <td>
                COURT OF APPEAL FILE NO.
              </td>
              <td>
                 
              </td>
            </tr>
            <tr>
              <td>
                <input
                  disabled={true}
                  value="Court of Appeal"
                />
              </td>
              <td>
                <input
                  autoFocus={true}
                  id="file-no"
                  name="number.form-seven"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  placeholder={undefined}
                  readOnly={undefined}
                  value="CA"
                />
              </td>
              <td>
                <button
                  className="btn btn-primary btn-green"
                  disabled={true}
                  id="find-button"
                  onClick={[Function]}
                >
                  Find
                </button>
              </td>
              <td>
                <div
                  className="error-message"
                >
                  
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="form-section"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="button-bar not-printable"
        >
          <button
            className="btn btn-warning"
            id="back"
            onClick={[Function]}
          >
            &lt; 
            Back
          </button>
          <button
            className="btn btn-success"
            id="draft"
            onClick={[Function]}
          >
            Save as Draft
          </button>
          <button
            className="btn btn-success"
            disabled={true}
            id="preview"
            onClick={[Function]}
          >
            Preview &gt;
          </button>
        </div>
      </div>
      <div
        className="modal"
        id="viewFormModal"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="modal-title green not-printable"
        >
          <span
            id="close-modal"
            onClick={[Function]}
          >
            ×
          </span>
          Preview Form 2
        </div>
        <div
          className="modal-content"
        >
          <div
            className="form-section"
          >
            <div
              className="button-bar not-printable"
            >
              <button
                className="btn btn-warning"
                id="back"
                onClick={[Function]}
              >
                &lt; 
                Back to editing
              </button>
              <button
                className="btn btn-success"
                id="draft"
                onClick={[Function]}
              >
                Save as Draft
              </button>
              <button
                className="btn btn-success"
                id="print"
                onClick={[Function]}
              >
                Print
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="modal not-printable"
        id="validationModal"
      >
        <div
          className="modal-title"
        >
          <span
            id="close-modal"
          >
            ×
          </span>
          Form Validation
        </div>
        <div
          className="modal-content"
        >
          <div
            className="progress"
          >
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="100"
              className="progress-bar progress-bar-striped"
              role="progressbar"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <span>
                100%
              </span>
            </div>
          </div>
          <table
            id="validations"
          >
            <tbody>
              <tr>
                <td
                  className="validation"
                >
                  All required fields filled in
                </td>
                <td>
                  <i
                    className="far fa-check-circle"
                  />
                </td>
              </tr>
              <tr>
                <td
                  className="validation"
                >
                  Address for Service is in British Columbia
                </td>
                <td>
                  <i
                    className="far fa-check-circle"
                  />
                </td>
              </tr>
            </tbody>
          </table>
          <hr />
          <div>
            <span
              className="success"
            >
              Your form has been successfully verified. Please proceed to review.
            </span>
            <button
              className="btn btn-primary"
            >
              Preview Form &gt;
            </button>
          </div>
        </div>
      </div>
      <div
        className="modal not-printable"
        id="saveErrorModal"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="modal-title red"
        >
          <span
            id="close-modal"
            onClick={[Function]}
          >
            ×
          </span>
          Save failed
        </div>
        <div
          className="modal-content"
        >
          <div>
            Something unexpected happened.
          </div>
        </div>
      </div>
      <div
        className="modal not-printable"
        id="saveSucessModal"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="modal-title green"
        >
          <span
            id="close-modal"
            onClick={[Function]}
          >
            ×
          </span>
          Saved!
        </div>
        <div
          className="modal-content"
        >
          <div>
            Saved as draft!
          </div>
        </div>
      </div>
      <div
        className="modal not-printable"
        id="dataLossWarning"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="modal-title warning"
        >
          <span
            id="close-modal"
            onClick={[Function]}
          >
            ×
          </span>
          Warning!
        </div>
        <div
          className="modal-content"
        >
          <div>
            All changes will be lost!
          </div>
          <div
            className="button-bar not-printable"
          >
            <button
              className="btn btn-warning"
              id="back"
              onClick={[Function]}
            >
              &lt; 
              Go back so I can save the form as draft.
            </button>
            <button
              className="btn btn-success"
              id="submit"
              onClick={[Function]}
            >
              Continue anyway.  I don't care about the data.
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
